components/infra-proxy-service/server/environments.go

```diff
@@ -143,7 +142,65 @@ func (s *Server) DeleteEnvironment(ctx context.Context, req *request.Environment
 	return &response.Environment{
 		Name: environment.Name,
 	}, nil
+}
+
+// UpdateEnvironment updates the environment attributes
+func (s *Server) UpdateEnvironment(ctx context.Context, req *request.UpdateEnvironment) (*response.Environment, error) {
+	c, err := s.createClient(ctx, req.OrgId, req.ServerId)
+	if err != nil {
+		return nil, err
+	}
+
+	if req.Name == "" {
```

Comment:
💭 Didn't we have a fancy validation lib for these kinds of things? 😉

Reply:
Yeah, As I started with https://github.com/chef/automate/pull/3540 and trying to add advance GPRS but observed major changes in UI, so suppressed it `NewPlainError` and pushed the PR.


I am tired of adding if condition lets have something better than this :relieved:

Reply:
I believe this is the library call that @srenatus is referring to:
https://github.com/chef/automate/blob/f65599c19f51bce014da776f8349d289e9e7a1d7/components/authz-service/server/project.go#L126-L129

Reply:
Yeah, @msorens we can utilize validation lib here.

but somehow, I also tried to add validation lib for infra service.

once PR #3540 will get merged. will replace it with something like

```
err := validation.New(validation.Options{
		Target:          "environment",
		Request:         *req,
		Rules: validation.Rules{
			"Name": []string{"required"},
		},
	}).Validate()


```

Reply:
I've meant yours, I thought that PR had been merged by now. :smiley: